http/getAxiosConfig.ts

```diff
@@ -67,19 +152,23 @@ export function getAxiosConfig(options: HttpOptions): AxiosRequestConfig {
     httpUseLocalhost = config.httpUseLocalhost;
   }
 
+  const baseURL = getHubSpotApiOrigin(
+    env,
+    localHostOverride ? false : httpUseLocalhost
+  );
+
   return {
-    baseURL: getHubSpotApiOrigin(
-      env,
-      localHostOverride ? false : httpUseLocalhost
-    ),
+    baseURL,
     headers: {
       ...getDefaultUserAgentHeader(),
       ...(headers || {}),
     },
     timeout: httpTimeout,
     transitional: DEFAULT_TRANSITIONAL,
-    httpAgent,
-    httpsAgent,
+    // Disable axios's built-in proxy handling - we use custom agents instead
+    proxy: false,
+    httpAgent: shouldUseProxy(baseURL) ? getHttpProxyAgent() || httpAgent : httpAgent,
```

Comment:
We are we doing a ternary and an or?  Shouldn't this just be `shouldUseProxy(baseURL) ? getHttpProxyAgent() : httpAgent`

I'm guessing it is to make TS happy, but we can probably refactor so that `getHttpProxyAgent` has a different name and returns either the `httpAgent` or the `HttpsProxyAgent`